src/test/java/seedu/address/model/person/GenderTest.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+
+public class GenderTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Gender(null));
+    }
+
+    @Test
+    public void constructor_invalidGender_throwsIllegalArgumentException() {
+        String invalidGender = "";
+        assertThrows(IllegalArgumentException.class, () -> new Gender(invalidGender));
+    }
+
+    @Test
+    public void isValidGender() {
+        //null Gender
+        assertThrows(NullPointerException.class, () -> Gender.isValidGender(null));
+
+        //invalid Gender
+        assertFalse(Gender.isValidGender("")); // empty string
+        assertFalse(Gender.isValidGender(" ")); // spaces only
+        assertFalse(Gender.isValidGender("T")); // only invalid characters
+
+        //valid Gender
+        assertTrue(Gender.isValidGender("M")); // for Male
+        assertTrue(Gender.isValidGender("F")); //for Female
+        assertTrue(Gender.isValidGender("m")); //for lowercase representation
+        assertTrue(Gender.isValidGender("f")); //for lowercase representation
```

Comment:
Good test cases for uppercase and lowercase

Reply:
I agree! good job on checking edge cases